src/ucs/sys/compiler_def.h

```diff
@@ -74,10 +74,10 @@
 #endif
 
 /* The i-th bit */
-#define UCS_BIT(i)               (1ul << (i))
+#define UCS_BIT(i)  (1ul << (i))
 
 /* Mask of bits 0..i-1 */
-#define UCS_MASK(i)              (UCS_BIT(i) - 1)
+#define UCS_MASK(i) ((i == 0) ? 0 : (UCS_BIT(i - 1) | (UCS_BIT(i - 1) - 1)))
```

Comment:
To make UCS_MASK support both 0 and 64 input value

Reply:
suggest to add parenthesis around `i`.

alternatively there is also `#define UCS_MASK(_i) ((((unsigned long long)!(_i))-1) >> (64 - (_i)))`, seems it _might_ be faster on micro benchmark (hpchead):
- -O2 and -O3: 3.38sec down to 3.10sec
- -O0: 8.71sec down to 7.27sec

Reply:
I wonder if we ever use UCS_MASK in a fast path. if yes, might consider adding UCS_MASK_SAFE macro instead of changing the existing macro

Reply:
maybe one macro to use is simpler since all fast paths need the safe version. what about renaming to something like UCS_BITSET(i), mask is usually not meant to be 0.

Reply:
Done.